airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/copy/s3/S3StreamCopier.java

```diff
@@ -181,15 +193,11 @@ public String generateMergeStatement(final String destTableName) {
   @Override
   public void removeFileAndDropTmpTable() throws Exception {
     if (purgeStagingData) {
-      for (final Map.Entry<String, DestinationFileWriter> entry : stagingWritersByFile.entrySet()) {
-        final String suffix = entry.getKey();
-        final String objectKey = entry.getValue().getOutputPath();
-
-        LOGGER.info("Begin cleaning s3 staging file {}.", objectKey);
-        if (s3Client.doesObjectExist(s3Config.getBucketName(), objectKey)) {
-          s3Client.deleteObject(s3Config.getBucketName(), objectKey);
+      for (final String fileName : stagingFileNames) {
+        if (s3Client.doesObjectExist(s3Config.getBucketName(), fileName)) {
```

Comment:
according to the AWS S3 SDK javadocs there is no need to check whether the file exists before delete

Reply:
it was old logic but i added this change)